rollup-operator/src/synch.js

```diff
@@ -274,6 +274,35 @@ class Synchronizer {
         await this.db.delete(`${eventOnChainKey}${separator}${batchNumber-1}`);
         // clear forge events
         await this.db.delete(`${eventForgeBatchKey}${separator}${batchNumber}`);
+        // clear exits batch
+        await this._clearExits(batchNumber);
+    }
+
+    async _clearExits(batchNumber){
+        const keysExits = await this.db.listKeys(`${exitInfoKey}${separator}`);
+        for (const exitIdKey of keysExits) {
+            const arrayExits = this._fromString(await this.db.get(exitIdKey));
+            this._purgeArray(arrayExits, batchNumber);
+            await this.db.insert(exitIdKey, this._toString(arrayExits));
+        }
+    }
+
+    _purgeArray(array, numBatch){
+        if (array.length === 0) return;
+        if (array.slice(-1)[0] < numBatch) return;
+        if (array[0] > numBatch) {
+            array.splice(0, array.length);
+            return;
+        }
+        let indexFound = null;
+        for (let i = array.length - 1; i >= 0; i--){
+            if (array[i] <= numBatch){
+                indexFound = i+1;
+                break;
+            }
```

Comment:
in case that ```array[i] = numbatch``` the indexFound should be 
```indesFound = i```

Example: batches array = [10], numBatch = 10

In that case indexFound should be 0, but will be 1.

Proposal:

```javascript
 if (array[i] <= numBatch){
               if(array[i] == numBatch)
                   indexFound = i;
               else
                   indexFound = i+1; //can't overflow the array cause we already check that (array.slice(-1)[0] > numBatch)
                break;
            }

```

Reply:
- If proposal is implemented we end up with an empty array such as:
`array = [10] -->  purgeArray(array, 10) --> array = []`
- Method has been implemented in order to remove array elements from `X` keeping element `X` in the array.
  - Examples:
`array = [10, 11, 12]` --> purgeArray(array, 11) --> array = [10, 11]`
`array = [10, 11, 12]` --> purgeArray(array, 10) --> array = [10]`
`array = [10, 11, 12]` --> purgeArray(array, 9) --> array = []`

If `indexFound` is above `array.length`, the array is not modified as it is said [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice):
`array = [10] -->  purgeArray(array, 10) --> array = [10]`

- This behavior implies that at the time to do `this._clearExits(batchNumber)` it should be `this._clearExits(batchNumber - 1)` in order to remove all batch numbers above roll-back number set.